### DISCUSSION TITLE
Add support for searching by Regex group (not by match)

### DISCUSSION THREAD

**Discussion by Pringlers** [zed-industries/zed#24617] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

Currently, when using search by Regex, Zed highlights the entire match.
It would be nice for Zed to support highlighting each group separately.

### If applicable, add mockups / screenshots to help present your vision of the feature

Current behavior (search by match):

![image](https://user-images.githubusercontent.com/98678118/217010480-80fd6eea-b910-414b-86e1-5bcfa9a6cdbc.png)

Suggested behavior (search by group):

<img width="698" alt="image" src="https://user-images.githubusercontent.com/98678118/217012345-419a4a46-b5e0-4921-ae32-38213a188ea3.png">

**Comment by notpeter** [comment]
How would you suggest that this behavior be activated?
Would searching for `foo(bar|baz)` default to only selecting `bar` or `baz` (this differs from how VSCode and others handle this case)? If so, would I have have to use `(foo(bar|baz))` to highlight the whole phrase?  
What happens if you have multiple capture groups in a single expression?
